lib/lib-storage/src/Upload.ts

```diff
@@ -81,19 +82,21 @@ export class Upload extends EventEmitter {
 
     // set defaults from options.
     this.queueSize = options.queueSize || this.queueSize;
-    this.partSize = options.partSize || this.partSize;
     this.leavePartsOnError = options.leavePartsOnError || this.leavePartsOnError;
     this.tags = options.tags || this.tags;
 
     this.client = options.client;
     this.params = options.params;
 
-    this.__validateInput();
-
     // set progress defaults
     this.totalBytes = byteLength(this.params.Body);
     this.bytesUploadedSoFar = 0;
     this.abortController = options.abortController ?? new AbortController();
+
+    this.partSize = this.__calculatePartSize(this.totalBytes ?? 0, Upload.MIN_PART_SIZE);
+    this.expectedPartsCount = this.totalBytes ? Math.ceil(this.totalBytes / this.partSize) : 1;
```

Comment:
I get the sense that this is a check for whether totalBytes was able to be calculated and not whether or not it is the numeric value 0.

In that case, it would be clearer to state the check as `this.totalBytes !== undefined ?`. However, with a zero byte input, the expected part count is indeed 1, so perhaps no change is needed.

----

But, why would expectedPartsCount be 1 if totalBytes could not be calculated? It could be a stream of an unknown byte length. In this case, both totalBytes and expectedPartCount should be `undefined`, since they would not be knowable.